CRM/Core/Payment.php

```diff
@@ -149,6 +149,338 @@ abstract class CRM_Core_Payment {
+
+  /**
+   * Payment processor generated string for the transaction ID.
+   *
+   * @var string
+   */
+  protected $transactionID;
+
+  /**
+   * Amount of money charged in fees by the payment processor.
+   *
+   * This is notified by (some) payment processers.
+   *
+   * @var float
+   */
+  protected $feeAmount;
+
+  /**
+   * Additional information returned by the payment processor regarding the payment outcome.
+   *
+   * This would normally be saved in civicrm_financial_trxn.result_code.
+   *
+   * @var string
+   */
+  protected $resultCode;
+
+  /**
+   * Combined with recurFrequencyUnit this gives how often billing will take place.
+   *
+   * e.g every 1 month
+   *
+   * @var int
+   */
+  protected $recurFrequencyInterval;
+
+  /**
+   * Combined with recurFrequencyInterval this gives how often billing will take place.
+   *
+   * e.g every 1 month
+   *
+   * @var string
+   *   month|day|year
+   */
+  protected $recurFrequencyUnit;
+
+  /**
+   * Passed in parameters.
+   *
+   * Using individual getters & setters is preferred but these will be used if
+   * they are not available.
+   *
+   * @var array
+   */
+  protected $inputParams = [];
+
+  /**
+   * @return int
+   */
+  public function getContactID(): int {
+    return $this->contactID ?? $this->inputParams['contactID'];
+  }
+
+  /**
+   * @param int $contactID
+   */
+  public function setContactID(int $contactID) {
+    $this->contactID = $contactID;
+  }
+
+  /**
+   * @return int
+   */
+  public function getContributionRecurID(): int {
+    return $this->contributionRecurID ?? $this->inputParams['contributionRecurID'];
+  }
+
+  /**
+   * @param int $contributionRecurID
+   */
+  public function setContributionRecurID(int $contributionRecurID) {
+    $this->contributionRecurID = $contributionRecurID;
+  }
+
+  /**
+   * @return string
+   */
+  public function getPaymentToken(): string {
+    return $this->paymentToken ?? $this->inputParams['token'];
+  }
+
+  /**
+   * @param string $paymentToken
+   */
+  public function setPaymentToken(string $paymentToken) {
+    $this->paymentToken = $paymentToken;
+  }
+
+  /**
+   * Get gateway generated transaction ID.
+   *
+   * @return string
+   */
+  public function getTransactionID(): string {
+    return $this->transactionID;
+  }
+
+  /**
+   * Set gateway generated transaction ID.
+   *
+   * @param string $transactionID
+   */
+  public function setTransactionID(string $transactionID) {
+    $this->transactionID = $transactionID;
+  }
+
+  /**
+   * Get additional result information received from gateway.
+   *
+   * This is saved to civicrm_financial_trxn.result_code.
+   *
+   * @return string
+   */
+  public function getResultCode(): string {
+    return $this->resultCode;
+  }
+
+  /**
+   * Set additional result information from gateway.
+   *
+   * This is saved to civicrm_financial_trxn.result_code.
+   *
+   * @param string $resultCode
+   */
+  public function setResultCode(string $resultCode) {
```

Comment:
See comment above - I'd prefer `get/setTrxnResultCode`

Reply:
OK - you win this one :-)